### ISSUE TITLE
Cache files are blank

### ISSUE THREAD

**Issue by Matan** [s3fs-fuse/s3fs-fuse#15]
Please look at this binary comparison image:
![image-differences](https://f.cloud.github.com/assets/184429/2228232/8b30ea2a-9ad7-11e3-88de-3329d439147f.JPG)
On the left we have the source image file downloaded from S3 console and on the right we have the cache file created by the s3fs mount.

Is this the expected behaviour? I noticed that the cache doesn't make a speed difference as the file is being downloaded from S3 each time. I tested latency to my web server in a couple of ways:
* Direct image load (no s3 mount): ~200 latency
* Direct image load (with mount, no cache): ~1000 latency
* Direct image load (with mount, with cache): ~1000 latency

This lead me to believe that the cache isn't working properly, thus I did the binary comparison and found the bytes of the source vs cache doesn't match up.

Urgent help would be much appreciated.

Thanks,
Matan

**Comment by ggtakec** [comment]
I'm sorry for that this issue had been left at a long period of time.

If you have this problem yet, please try to use latest code.

Thanks in advance for your help.

**Comment by Morc001** [comment]
Hi Takeshi,

  This problem of files containing only zeros (empty sparse files?) is still happening on master as of today. Files are small, a few hundred bytes only, so no multi-part. I'm just mounting a bucket and viewing files, nothing fancy. I'm using eu-central-1 endpoint if it makes any difference. In your last comment you wrote it's fixed or has that referred to some other problem?
  Using 1.79 the problem does not show, but that version lacks ensure_diskfree.

**Comment by davrodfer** [comment]
I had the same issue. I wrote my files from two different machines with the same role. When I read a file from a machine that didn't wrote the file, all I have is a bunch of zeroes. Every machine only can read the files that they wrote, but the files are Ok in S3.

Y had this configuration in /etc/fstab:
```
s3fs#s3bucketname  /mnt/s3fs/s3bucketname     fuse    rw,nosuid,nodev,allow_other,uid=48,gid=48,use_cache=/var/tmp/s3fs,default_acl=public-read-write,storage_class=standard,_netdev,use_rrs,iam_role=role_s3bucketname 0 0
```
I changed to:
```
s3fs#s3bucketname  /mnt/s3fs/s3bucketname     fuse    noatime,rw,nosuid,nodev,allow_other,uid=48,gid=48,default_acl=public-read-write,storage_class=standard,_netdev,iam_role=role_s3bucketname 0 0
```
And now it works, I can read the files from all machines. I have disabled the cache. I don't know if this can be a problem in the future.

**Comment by aggronerd** [comment]
I am getting the same. Files appearing to have zero bits. This has only just happened recently (last couple of weeks) so I have got around the issue by rolling back to SHA cf56b35766c4450c789968335ffbc01258cbbdb8. I haven't really been able to debug it further and can't afford to lose caching.

The volume in my case is mounted from the command line:

`s3fs $BUCKET_NAME $MOUNTPOINT -o iam_role=$IAM_ROLE,nosuid,nonempty,nodev,use_cache=/tmp,allow_other,retries=5,url=https://s3-eu-west-1.amazonaws.com/`

**Comment by gl-lamhnguyen** [comment]
Yes this happened recently. What's the fix?

**Comment by ggtakec** [comment]
@Morc001 @davrodfer @gl-lamhnguyen 
I'm sorry for replying late.
As mentioned by aggronerd, I've made the bug.
I try to fix this, please wait a while.
Regards,

**Comment by CloudaYolla** [comment]
Hi @ggtakec,
We are using s3fs in a project, and suddenly everything started not working. After a week of investigation (searching the problem elsewhere), today we found out that the existing files mounted from s3fs are shown corrupt. But when a new file is created, it is written and read ok in the s3 mount folder.

Is there any workaround for this? I didn't understand the thread below if there is any? 
We cannot afford to wait for a fix to come.

Thanks and best regards,
EP Team

**Comment by gl-lamhnguyen** [comment]
@CloudaYolla I guess your broken s3fs was pulled from latest commit. I couldn't wait either, so I used this SHA instead https://github.com/s3fs-fuse/s3fs-fuse/tree/cf56b35766c4450c789968335ffbc01258cbbdb8

Download ZIP and run install commands, things will be back to normal. Lesson learned, we should keep a copy of a working version on our end at all time.

**Comment by ggtakec** [comment]
@CloudaYolla @gl-lamhnguyen 
I reverted #379 on master, so please use d048f38.
And I wil fix a bug(#379) later.
Thanks in advance for your help.

**Comment by ggtakec** [comment]
#395 merged on master branch, please try to use latest codes.
Thanks in advance for your help.

**Comment by CloudaYolla** [comment]
Hi @ggtakec,
So this means, we don't need to revert to the mentioned previous version?

Thank you,
EP Team

**Comment by ggtakec** [comment]
@CloudaYolla 
Since This bug had been affecting most users, I needed to revert the code until fixing it.
Then this bug was fixed by #395 which patched for #379.

I am sorry for my fault.
I closed this issue, if you found a same bug, please reopen this.
Regards,
